retrieval/discovery/marathon/marathon_test.go

```diff
@@ -27,7 +27,7 @@ import (
 var marathonValidLabel = map[string]string{"prometheus": "yes"}
 
 func newTestDiscovery(client AppListClient) (chan []*config.TargetGroup, *Discovery) {
-	ch := make(chan []*config.TargetGroup)
+	ch := make(chan []*config.TargetGroup, 1)
```

Comment:
At least they will reliably block, not randomly...

The semantics is to do concurrent reading if you want to put more than one item in, and you can do it all sequentially if you use only one item. I find that way easier to reason with than concurrent contraption in every single test.